lib/heap_sort.rb

```diff
@@ -1,8 +1,50 @@
 
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
+# Time Complexity:  O(n log n)
+# Space Complexity: O(1)
 def heap_sort(list)
```

Comment:
👍 , Nice work doing this in-place with heapsort.  Using the given array as the heap.  Well done.